src/sinol_make/commands/gen/__init__.py

```diff
@@ -0,0 +1,147 @@
+import argparse
+import glob
+import os
+import subprocess
+import hashlib
+import yaml
+
+import multiprocessing as mp
+
+from sinol_make import util
+from sinol_make.commands.gen import gen_util
+from sinol_make.commands.gen.structs import OutputGenerationArguments
+from sinol_make.helpers import parsers, package_util, compile
+from sinol_make.interfaces.BaseCommand import BaseCommand
+
+class Command(BaseCommand):
+    """
+    Class for `gen` command.
+    """
+
+    def get_name(self):
+        return "gen"
+
+    def configure_subparser(self, subparser):
+        parser = subparser.add_parser(
+            self.get_name(),
+            help='Generate input and output files',
+            description='Generate input files using gingen program '
+                        '(for example prog/abcingen.cpp for abc task). If there '
+                        'is a new test or it differs from the previous one, '
+                        'then the output file will be generated using the '
+                        'correct solution. You can also specify your ingen source '
+                        'file which will be used.'
+        )
+
+        parser.add_argument('ingen_path', type=str, nargs='?',
+                            help='path to ingen source file, for example prog/abcingen.cpp')
+        parser.add_argument('-c', '--cpus', type=int,
+                            help=f'number of cpus to use, by default {mp.cpu_count()} (all available). '
+                                 f'Used when generating output files.', default=mp.cpu_count())
+        parsers.add_compilation_arguments(parser)
+
+    def compile_ingen(self):
+        self.ingen_exe, compile_log_path = gen_util.compile_ingen(self.ingen, self.args,
+                                                                  self.args.weak_compilation_flags)
+        if self.ingen_exe is None:
+            print(util.error('Failed ingen compilation.'))
+            compile.print_compile_log(compile_log_path)
+            exit(1)
+        else:
+            print(util.info('Successfully compiled ingen.'))
```

Comment:
This `if` looks like can be put into the `gen_util.compile_ingen` function. Is there any reason why error handling is here, instead of the function's definition?

Reply:
I moved it there